### DISCUSSION TITLE
Wordpress plugin: add support for <html lang="">

### DISCUSSION THREAD

**Discussion by Sapper-Morton** [mosparo/mosparo#230] [Integrations]
Currently it seems that Mosparo detects language from device locale. This won't always get best results because people use different language on their devices and the site where Mosparo is used might be in another one. Adding an option to Mosparo Wordpress settings that Mosparo would prefer html lang tag first and if there's none, use device language. This might be usable on wider range too...

**Comment by zepich** [comment]
Hi @Sapper-Morton 

Thank you very much for your suggestion.

That's right, mosparo uses the device locale for maximum accessibility. The idea is that the user can understand the checkbox as well as possible, even if the website is not in the user's language.

However, I understand that in some cases, the different languages on the same page do not look good. Because of this, it's possible to define the language in the JavaScript code included in your website (https://documentation.mosparo.io/docs/integration/custom#additional-options, see `language`).

To use this option with WordPress, you must use a filter to set the language. You can use the filter `mosparo_integration_filter_frontend_options` to specify the options you want (See here:
https://github.com/mosparo/wordpress-plugin/blob/master/src/MosparoIntegration/Helper/FrontendHelper.php#L111).

Something like this should work:

```php
function mosparoOverrideOptions($options)
{
    $options['language'] = 'fi_FI'; // Specify the language you want

    return $options;
}
add_filter('mosparo_integration_filter_frontend_options', 'mosparoOverrideOptions');
```

Users will now see the mosparo box in the locale `fi_FI`. Depending on your multilanguage plugin, you can also use the function to get the current language (for example, `pll_current_language()` if you use Polylang).

With that option in mind, do you still think the mosparo WordPress plugin should be able to use the language in the `html` tag?

Kind regards,

zepich

**Reply by Sapper-Morton** [reply]
Hi,

thanks for the answer.
For simplicity, I'd see a checkbox in the wordpress plugin's settings which would be:
[X] Prefer html lang over device language

The less your users need to do custom code, the better.

**Reply by zepich** [reply]
Hi @Sapper-Morton 

Thank you very much for your feedback.

I agree with you. I had some trouble finding a good solution for such an option, but after thinking about your suggestion, I think I found one.

Instead of adding this functionality to the plugin, we'll add it to mosparo directly. The problem with adding it to the plugin is that we would have to add such an option to all plugins and integrations, which is a lot of work.

So, my suggestion is that the frontend script always sends the HTML language to mosparo's frontend API.

In the project settings in mosparo, the project owner can choose between the following options (per project):
- Use browser language, fallback to English (standard, as it is now)
- Use browser language, fallback to HTML tag language, fallback to English
- Use HTML tag language, fallback to Browser language, fallback to English

It will always be possible to override the language with the option in the frontend script.

What do you think about such an implementation?

Thank you very much for your opinion.

Kind regards,

zepich

**Reply by Sapper-Morton** [reply]
Hey,

Sorry for the huge delay. This seems really good :)

Thanks

**Comment by zepich** [comment]
Added with v1.3
